jujugui/static/gui/src/app/app.js

```diff
@@ -2110,6 +2109,21 @@ YUI.add('juju-gui', function(Y) {
       next();
     },
 
+    _ensureControllerConnection: function(state, next) {
+      if (
+        state.root === 'logout' ||
+        this.controllerAPI.get('connecting') ||
+        this.controllerAPI.get('connected') ||
+        this.controllerAPI.userIsAuthenticated) {
+        console.log('ignoring controller connection');
+        next();
+        return;
+      }
+      console.log('controller connecting');
```

Comment:
The `connect` method in the controllerAPI already logs this out, probably don't need to duplicate it.